XamarinExam/Controllers/DataManager.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using ConsoleTables;
+using Newtonsoft.Json;
+using XamarinExam.Models;
+
+namespace XamarinExam.Controllers
+{
+    public class DataManager
+    {
+
+        private static DataManager instance;
+
+        public List<Student> Students { get; set; }
```

Comment:
Create a manager inteface
```C#
interface IManager{
   void Input();
   void Save();
   void Read();
}
```

Make a base class for all model manager like
```C# 
public class BaseManager<T> : List<T>, IManager where T: DataModel
{

     protected virtual void Input()
    {
      Console.WriteLine("Nhap danh sach XXXX.");///XXX = read from annotation of T
      /// Your code
     }
     public void Save()
     {
        //your code
     }
     public void Load()
     {
       // your code
     }
    /// Other common methods
.....
}
```
Then
```C#
public class StudentManager : BaseManager<Student>
{
    public double CalculateAverageScore(int studentId)
   {
      //your code
   }
}
```
From DataManager
```C#
public class DataManager//Make singleton, 
{
    private List<IManager> managers = new List<IManager>(){
      new StudentManager(),
      new ScoreManager(),
      new .....
   }
   public BaseManager<T> ForType<T>() where T: BaseModel
   {
      return manager.OfType<BaseManager<T>>().First();
   }
   ///Your other ultility methods
  public void SaveAll()
  {
      managers.ForEach....
   }
}

Then you can call
DataManager.Instance.ForType<Student>().Input();
DataManager.Instance.SaveAll();
DataManager.Instance.LoadAll();
```